app/src/main/java/app/revanced/integrations/sponsorblock/PlayerController.java

```diff
@@ -45,6 +45,10 @@ public class PlayerController {
 
     public static boolean shorts_playing = false;
 
+    public static void shortsOpened() {
+        shorts_playing = true;
+    }
+
```

Comment:
Previously the field `shorts_playing` was set directly by injecting the necessary instructions for that in the method when shorts open. Since this PR now makes use of a patch to detect if shorts open, this method was introduced instead. It was found that `shorts_playing` is never set back to false, which blocks `executeDownloadSegments`. I am not sure though, why it still works.

Reply:
shorts_playing is set to false using borrowed code from the autocaptions patch:

```
        val startVideoInformerMethod = StartVideoInformerFingerprint.result!!.mutableMethod
        startVideoInformerMethod.addInstructions(
            0, """
            const/4 v0, 0x0
            sput-boolean v0, $INTEGRATIONS_PLAYER_CONTROLLER_CLASS_DESCRIPTOR->shorts_playing:Z
        """
        )
```
Although this could be removed.  Since VideoIdPatch does not support shorts, the call to PlayerController#setCurrentVideoId could detect when a non short is loaded.

Reply:
> Although this could be removed. Since VideoIdPatch does not support shorts, the call to PlayerController#setCurrentVideoId could detect when a non short is loaded.

I do not understand this quite yet. Does this snippet not already suffice to determine if shorts are playing? If that is the case, `VideoInformationPatch` should be refactored to include a getter for when shorts are playing. This would allow other patches to depend on it to check if shorts are currently playing such as the patch for RYD and SponsorBlock, since SponsorBlock does not properly integrate with shorts. This refactoring change would get rid of the following code from the SponsorBlock patch:
 
```kt
        val startVideoInformerMethod = StartVideoInformerFingerprint.result!!.mutableMethod
        startVideoInformerMethod.addInstructions(
            0, """
            const/4 v0, 0x0
            sput-boolean v0, $INTEGRATIONS_PLAYER_CONTROLLER_CLASS_DESCRIPTOR->shorts_playing:Z
        """
        )

        val shortsPlayerConstructorMethod = ShortsPlayerConstructorFingerprint.result!!.mutableMethod

        shortsPlayerConstructorMethod.addInstructions(
            0, """
            const/4 v0, 0x1
            sput-boolean v0, $INTEGRATIONS_PLAYER_CONTROLLER_CLASS_DESCRIPTOR->shorts_playing:Z
        """
        )
```

Instead, SponsorBlock would depend on `VideoInformationPatch` and use its getter for `shortsPlaying`.

Reply:
Yes that sounds correct.

Additional, since VideoIdPatch does not currently support shorts, that means sponsorblock does not need to check if a short is playing since it will only be notified of regular video ids that are loaded.

So the two patch injections you quoted could be removed from sponsorblock without any problem.

Reply:
I am still confused why the `sponsorblock` patch injects code which sets `shorts_playing` to true or false in addition to the current shorts detection patch. Instead of the current approach, the VideoInformationPatch class should adopt that code from the `sponsorblock` patch and provide a getter for getting the current state of shorts playing. As explained, this would get rid of introducing the ShortsDetectionPatch class. Instead other patches would use VideoInformationPatch as a dependency so that their integration can make use of said getter. After all, it makes sense that shorts playing state is related to video information, so it belongs in that class.

Reply:
I agree

Reply:
I agree that shorts detection should be moved into VideoInformationPatch.